src/main/drivers/exti.c

```diff
@@ -176,40 +176,78 @@ void EXTIRelease(IO_t io)
     // don't forget to match cleanup with config
     EXTIEnable(io, false);
 
-    int chIdx;
-    chIdx = IO_GPIOPinIdx(io);
-    if (chIdx < 0)
+    int chIdx = IO_GPIOPinIdx(io);
+
+    if (chIdx < 0) {
         return;
+    }
+
     extiChannelRec_t *rec = &extiChannelRecs[chIdx];
     rec->handler = NULL;
 }
 
 void EXTIEnable(IO_t io, bool enable)
 {
 #if defined(STM32F1) || defined(STM32F4) || defined(STM32F7)
+
     uint32_t extiLine = IO_EXTI_Line(io);
-    if (!extiLine)
+
+    if (!extiLine) {
         return;
-    if (enable)
+    }
+
+    if (enable) {
         EXTI->IMR |= extiLine;
-    else
+    } else {
         EXTI->IMR &= ~extiLine;
+    }
+
+#elif defined(STM32H7)
+
+    uint32_t extiLine = IO_EXTI_Line(io);
+    if (!extiLine) {
+        return;
+    }
+
+    if (enable) {
+        EXTI_D1->IMR1 |= extiLine;
+    } else {
+        EXTI_D1->IMR1 &= ~extiLine;
+    }
+
 #elif defined(STM32F303xC)
+
     int extiLine = IO_EXTI_Line(io);
-    if (extiLine < 0)
+
+    if (extiLine < 0) {
         return;
+    }
+
     // assume extiLine < 32 (valid for all EXTI pins)
-    if (enable)
+
+    if (enable) {
         EXTI->IMR |= 1 << extiLine;
-    else
+    } else {
         EXTI->IMR &= ~(1 << extiLine);
+    }
 #else
 # error "Unknown CPU"
 #endif
 }
 
 void EXTI_IRQHandler(void)
 {
+#ifdef STM32H7
+    uint32_t exti_active = EXTI_D1->IMR1 & EXTI_D1->PR1;
```

Comment:
It may be better to `#define` `IMR` and `PR1` and reuse code (also in EXTIEnable)
Just keep bitmask and bit index case

Reply:
Yeah, I thought the same thing for a moment, but hesitated to do so because a short macro like `#define IMR IMR1` would collide with something else already or will be defined outside the file, probably in library files.

Reply:
`#define EXTI_REG_IMR (EXTI_D1->IMR1)` ?

Reply:
What about defining EXTI_REG_IMR and EXTI_REG_PR around this function / in this file? The code duplication is annoying